maven-release-manager/src/main/java/org/apache/maven/shared/release/phase/InputVariablesPhase.java

```diff
@@ -159,9 +167,18 @@ public ReleaseResult execute( ReleaseDescriptor releaseDescriptor, ReleaseEnviro
             {
                 try
                 {
-                    tag =
-                        prompter.prompt( "What is SCM release tag or label for \"" + project.getName() + "\"? ("
-                            + project.getGroupId() + ":" + project.getArtifactId() + ")", defaultTag );
+                    if ( branchOperation )
+                    {
+                        tag =
+                            prompter.prompt( "What is the branch name for \"" + project.getName() + "\"? ("
+                                             + project.getGroupId() + ":" + project.getArtifactId() + ")", defaultTag );
```

Comment:
Instead of defaulting to the calculated tag name when prompting for the branch name, it would be better to fail if the user inputs an empty line. The default will almost never be the right one, and it would align with the current behaviour: for example, in batch mode without a branch name, the plugin should fail.

Additionally, this could be refactored in a method `getScmReleaseLabel`, returning the release label to use based on the default tag and release descriptor, for clarity.